ttnn/cpp/ttnn/operations/unary.hpp

```diff
@@ -102,11 +104,34 @@ struct Exp : public EltwiseUnary {
             memory_config);
     }
 };
+
+struct Softplus : public EltwiseUnary {
+    static const std::array<TensorSchema, 1> input_tensor_schemas() { return detail::input_tensor_schemas(); }
+
+    template <typename... Args>
+    static auto input_tensors_to_validate(const Tensor& input_tensor, Args&&... args) {
+        return detail::input_tensors_to_validate(input_tensor, std::forward<Args>(args)...);
+    };
+
+    static Tensor execute(const Tensor& input, const float beta, const float threshold, const std::optional<MemoryConfig>& memory_config_arg = std::nullopt) {
+        auto original_input_shape = input.get_shape();
+        auto input_4D = ttnn::unsqueeze_to_4D(input);
+
```

Comment:
Is it possible to get rid of the unsqueeze_to_4D?

Reply:
@eyonland , it may be possible if we add validation and rewrite tests to use 4D tensors. Underlying operations only support 4D right now. Keeping the change minimal, migrating from py to c++ for now. 

I will consult with @arakhmati whether we want to enforce on the contract level.

Reply:
Also, to clarify, this is not an addition. This is moved from py to c++.